content/attacks/posts/2023-02-07-CoW-Swap.md

```diff
@@ -0,0 +1,43 @@
+---
+date: 2023-02-07
+target-entities: 
+- CoW Swap
+entity-types:
+- DeFi
+- Exchange
+attack-types:
+- Smart Contract Exploit
+title: "CoW Swap Suffers Smart Contract Exploit Resulting in Approximately $166K Loss"
+---
+
+## Summary
+
+On February 7, 2023, CoW Swap, a decentralized exchange (DEX) protocol, fell victim to a smart contract exploit, resulting in a loss of approximately 550 BNB or about $180,000 USD. The breach occurred due to a flaw in the protocol's smart contract, allowing an unidentified attacker to approve fund transfers from the protocol.
+
+## Attackers
+
+The identity of the attacker is unknown.
+
+- [0xc0e82c1ed4786f8b7f806d1b8a6335ec485266ff](https://etherscan.io/address/0xc0e82c1ed4786f8b7f806d1b8a6335ec485266ff)
+- [0x55a37a2e5e5973510ac9d9c723aec213fa161919](https://etherscan.io/address/0x55a37a2e5e5973510ac9d9c723aec213fa161919)
+
+## Losses
+
+- $166,183
+
+## Timeline
+
+- **January 27, 2023:** Barter solver enters the CoW Swap solver competition. After being [allowlisted](https://etherscan.io/tx/0x7374b8157267b0c6d8cc05e812e68ddff045c5bdb5ca915c32193e9f752d5947), they [approve](https://etherscan.io/tx/0x92f906bce94bab417cccc87ae046448d7fb8c2c0350b7ed911545577acb3bfc1) their SwapGuard contract.
+- **February 7, 2023:** Attackers [exploit](https://dune.com/queries/1543508) a vulnerability in the SwapGuard contract to transfer funds from the CoW Swap's settlement contract to their accounts.
+- **February 7, 2023:** CoW Swap and Barter teams mitigate further damage by identifying the vulnerability, revoking all approvals for the vulnerable contract, and updating the Barter Solver contract.
+- **February 8, 2023:** Barter solver [refunds](https://etherscan.io/tx/0x61a5196468b276bcedc3aab2f2fe255c2d54ca1c655523f4b25c99b1a224e715) the losses caused by the hack.
+
+[source - CoW Swap forum](https://forum.cow.fi/t/cip-22-slashing-of-the-barter-solver-responsible-for-a-hack-causing-cow-dao-a-loss-of-1-week-fee-accrual/1440)
+
+## Security Failure Causes
+
+Several reasons according to the CoW Swap [report](https://blog.cow.fi/cow-swap-solver-exploit-post-mortem-07-02-2023-2faa9f918e29):
+- **Arbitrary Execution:** The SwapGuard contract developed by the Barter solver had a critical flaw. It allowed arbitrary execution of calls, a feature that the attackers exploited to drain tokens.
+- **Lack of Contract Validation:** The GPv2Settlement contract, part of the CoW Swap protocol, did not validate the token received in the settle() function, allowing the attackers to force it to approve a malicious contract.
+- **Unrestricted Approvals:** The Barter solver approved the vulnerable SwapGuard contract with a maximum value of DAI without adequately securing the contract against potential exploits.
+- **Lack of Access Control:** The smart contracts did not have properly implemented access control measures, which exposed sensitive functions externally.
```

Comment:
Remove repeating point, since `Arbitrary Execution` is already pointed, and the meaning is the same